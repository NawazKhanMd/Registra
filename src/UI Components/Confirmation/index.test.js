import React from 'react';
import renderer from 'react-test-renderer';
import { Confirmation } from './index';
import '@testing-library/jest-dom/extend-expect';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));
jest.mock('react-router', () => ({
  useHistory: () => ({
    push: jest.fn(),
    location: { pathname: '/' }
  }),
}));
jest.mock('firebase', () => ({
  initializeApp: () => ({
    analytics: jest.fn(),
    auth: jest.fn(),
  }), firestore: jest.fn(),
}));

jest.mock('react-redux', () => {
  const ActualReactRedux = require.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return 1;
    }),
    useDispatch: () => jest.fn()
  };
});


test('Confirmation Loads', () => {
  const component = renderer.create(
    <Confirmation />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Simulating Form submit', () => {
  const wrapper = shallow((<Confirmation />));
  wrapper.find('.red').simulate('click');
  wrapper.find('#reset').simulate('click');
});
test('Simulating Back', () => {
  const wrapper = shallow((<Confirmation />));
  wrapper.find('.grey').simulate('click');
});

