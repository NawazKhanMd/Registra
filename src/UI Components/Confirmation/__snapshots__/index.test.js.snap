// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Loads 1`] = `
<div
  className="confirm_container"
>
  <div
    className="neo_card"
  >
    <div
      className="d-flex f_row justify-content-evenly f-wrap"
    >
      <div
        className="image_container"
      >
        <input
          className="d-none"
          onChange={[Function]}
          type="file"
        />
        <img
          src="https://cdn2.iconfinder.com/data/icons/action-states-vol-4-flat/48/Action___States_-_Vol._4-10-512.png"
        />
        <h3
          className="img_btn"
        >
          <span
            onClick={[Function]}
          >
            ðŸ“‚
          </span>
          <span
            onClick={[Function]}
          >
            ðŸ“·
          </span>
        </h3>
      </div>
      <div
        className="segment seperator"
      >
        <p>
          <span>
            Name:
          </span>
           
        </p>
        <p>
          <span>
            Phn Number:
          </span>
           
        </p>
        <p>
          <span>
            Email:
          </span>
           
        </p>
        <p
          className="margin0"
        >
          <span>
            Address:
          </span>
          <span
            className="text-right"
          >
             
            , 
          </span>
        </p>
        <p
          className="justify-content-end"
        />
      </div>
    </div>
    <div
      className="segment"
    >
      <p>
        <span>
          Building Name:
        </span>
         
      </p>
      <p>
        <span>
          City/Area:
        </span>
         
      </p>
      <p>
        <span>
          Landline:
        </span>
         
      </p>
      <p
        className="margin0"
      >
        <span>
          Address:
        </span>
        <span
          className="text-right"
        >
           
          , 
        </span>
      </p>
      <p
        className="justify-content-end"
      />
    </div>
  </div>
  <div
    className="neo_card margin_t_20"
  >
    <canvas
      height="160"
      id="sig-canvas"
      width="620"
    >
      Your browser does not Suppport this feature
    </canvas>
    <div
      className="d-flex f-cloumn justify-content-center"
    >
      <div
        className="col-md-12"
      >
        <small>
          Sign in the canvas and save your signature!
        </small>
      </div>
      <div
        className="col-md-12 d-flex f-cloumn align-items-center"
      >
        <button
          className="red margin-10"
          id="sig-submitBtn"
        >
          Save Signature
        </button>
        <button
          className="grey margin-10"
          id="sig-clearBtn"
        >
          Clear Signature
        </button>
      </div>
    </div>
  </div>
  <div
    className="d-flex f_row justify-content-center margin-10"
  >
    <button
      className="grey  margin-10"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="red  margin-10"
      onClick={[Function]}
      type="submit"
    >
      Confirm
    </button>
  </div>
</div>
`;
